app/services/katello/repository_type_manager.rb

```diff
@@ -0,0 +1,59 @@
+module Katello
+  class RepositoryTypeManager
```

Comment:
All I can say here is that it follows the precedent set here ->  https://github.com/theforeman/foreman/blob/develop/app/services/foreman/plugin.rb#L34

Would renaming RepositoryTypeManager to RepositoryType help resolved your point ?